k-distribution/tutorial/2_languages/4_logik/basic/logik.k

```diff
@@ -343,7 +343,7 @@ another clause can be chosen using the rule above):  */
   rule <k> L:Term => . ...</k>
        <fresh>  L:Term . => . </fresh>
 
-  rule <k> L:Term => Ls ...</k>
+  rule <k> L:Term :KItem => Ls ...</k>
```

Comment:
The reason this happens is in order to set the sort of the rewrite to KItem instead of Terms. This then prevents a list terminator from being erroneously added.

Reply:
@radumereuta I agree that it looks suspicious, but I think it is ok.  Now with the more generous AML semantics to K, where you can define and do pretty much whatever makes sense with sorts and types, almost every decision that makes practical sense is acceptable.  Especially if it improves the performance and the error messages.